### PR TITLE
EmuThread: refactor

### PR DESCRIPTION
**Pull request by liamwhite**
This gets rid of some spaghetti surrounding starting and stopping emulation.

### CODE REVIEW

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -3901,79 +3900,6 @@ void GMainWindow::OnMouseActivity() {
     mouse_center_timer.stop();
 }
 
-void GMainWindow::OnCoreError(Core::SystemResultStatus result, std::string details) {
```
Comment (FearlessTobi) [inline src/yuzu/main.cpp:3904]: Why is this removed?

Comment (liamwhite) [reply src/yuzu/main.cpp:3904]: It isn't possible to reach.

**Review by Morph1984 — APPROVED** [review]
